### PR TITLE
ref(ts): Convert Hovercard to typescript

### PR DESCRIPTION
**Pull request by matejminar**

### CODE REVIEW

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/components/hovercard.tsx [hunk 1]
```diff
@@ -191,12 +216,17 @@ const slideIn = p => keyframes`
...
 
-const StyledHovercard = styled('div')`
+const getOffset = (p: StyledHovercardProps) => p.offset ?? space(2);
+
+type StyledHovercardProps = {
+  visible: boolean;
+  placement: Direction;
+  offset?: string;
+};
+const StyledHovercard = styled('div')<StyledHovercardProps>`
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/hovercard.tsx:229]:
```suggestion
};

const StyledHovercard = styled('div')<StyledHovercardProps>`
```

File: src/sentry/static/sentry/app/components/hovercard.tsx [hunk 2]
```diff
@@ -191,12 +216,17 @@ const slideIn = p => keyframes`
...
   border-radius: 4px;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/hovercard.tsx:230]:
```suggestion
  border-radius: ${p => p.theme.borderRadius};
```

File: src/sentry/static/sentry/app/components/hovercard.tsx [hunk 3]
```diff
@@ -228,24 +258,22 @@ const StyledHovercard = styled('div')`
 `;
 
 const Header = styled('div')`
-  font-size: 14px;
+  font-size: ${p => p.theme.fontSizeMedium};
   background: ${p => p.theme.gray100};
   border-bottom: 1px solid ${p => p.theme.borderLight};
   border-radius: 4px 4px 0 0;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/hovercard.tsx:264]:
```suggestion
  border-radius: ${p => p.theme.borderRadiusTop};
```

File: src/sentry/static/sentry/app/components/hovercard.tsx [hunk 4]
```diff
@@ -228,24 +258,22 @@ const StyledHovercard = styled('div')`
...
 `;
 
 const Body = styled('div')`
   padding: ${space(2)};
   min-height: 30px;
 `;
 
-const HovercardArrow = styled('span')`
+type HovercardArrowProps = {placement: Direction; tipColor?: string};
+const HovercardArrow = styled('span')<HovercardArrowProps>`
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/hovercard.tsx:276]:
```suggestion
type HovercardArrowProps = {placement: Direction; tipColor?: string};

const HovercardArrow = styled('span')<HovercardArrowProps>`
```

**Review by priscilawebdev — APPROVED** [review]
I left some suggestions, but it looks good :)

**Pull request merged**